### ISSUE TITLE
Lunch program with GUI

### ISSUE THREAD

**Issue by adrienb4** [kohsuke/winsw#230]
Hi :)

I have correctly confirgured and started my program with Winsw, but the GUI is not displayed on the screen.

Do you have any options for this ?

Thank

Adrien

**Comment by oleg-nenashev** [comment]
@adrienb4 see https://github.com/kohsuke/winsw/blob/d192c03638565a53b92d8ad1fbb8d9d19ca6ae28/doc/xmlConfigFile.md#interactive

**Comment by adrienb4** [comment]
@oleg-nenashev : thank for your reply.

Sorry but i already use this option (i dit not specify it in my issue) and my GUI is not displayed :/

Adrien

**Comment by NextTurn** [comment]
@oleg-nenashev Here are some configuration design questions:

- Does it make sense to start the service in one account but the wrapped process in another account?
`<interactiveaccount/>`
- Does it make sense to start `startexecutable` and `stopexecutable` in different accounts?
`<startinteractive/>`/`<stopinteractive/>`
- Shall we provide a util to start a child process interactively? The service (typically a CLI) can run with privileges while the service controller (typically a GUI) interacts with the user.
IPC will be involved.

**Comment by CampeloMarcos** [comment]
https://docs.microsoft.com/en-us/windows/win32/services/interactive-services

**Comment by NextTurn** [comment]
> Would the solution suggested by them solve the problem?

Yes, if you are referring to *Interacting with a User from a Service Indirectly*.

We just need to design the configuration format. Any thoughts?

**Comment by CampeloMarcos** [comment]
With the existing <interactive /> tag I would keep the "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Windows\NoInteractiveServices" record to 0

and implement as recommended:
"To create an interactive service, do the following when calling the CreateService function:

Specify NULL for the lpServiceStartName parameter to run the service in the context of the LocalSystem account.
Specify the SERVICE_INTERACTIVE_PROCESS flag.
To determine whether a service is running as an interactive service, call the GetProcessWindowStation function to retrieve a handle to the window station, and the GetUserObjectInformation function to test whether the window station has the WSF_VISIBLE attribute. "

**Comment by hellodword** [comment]
Any updates? I got the same issue with a GUI program.

**Comment by Lauwisme** [comment]
I would like to use this feature as well, what is the current status?

**Comment by defnotmichal** [comment]
I'm also looking forward to this, although I hardly see a solution to this. Good luck to us Jenkins folks :)
Might start looking into Azure Pipelines and how they run those things in interactive mode - because they do somehow - maybe that'd be the proper way to work with this?
